Added Remove Debug Output Option

TheFocusMan opened the pull request:
If the option enabled the compiler run faster,
The fast boost also remove more than 1000 lines of code in the final asm file

TheFocusMan commented:
> This should have a different name other than fast boost, it's not proper English

The English matter?

TheFocusMan commented:
> > > This should have a different name other than fast boost, it's not proper English
> > 
> > 
> > The English matter? if the function is work
> 
> Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt

I dont have a better way to name this

terminal-cs commented:
> > > > This should have a different name other than fast boost, it's not proper English
> > > 
> > > 
> > > The English matter? if the function is work
> > 
> > 
> > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> 
> I dont have a better way to name this

name it EnableOptimizations or something

TheFocusMan commented:
> > > > > This should have a different name other than fast boost, it's not proper English
> > > > 
> > > > 
> > > > The English matter? if the function is work
> > > 
> > > 
> > > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> > 
> > 
> > I dont have a better way to name this
> 
> name it EnableOptimizations or something
I renamed it

terminal-cs commented:
> > > > > > This should have a different name other than fast boost, it's not proper English
> > > > > 
> > > > > 
> > > > > The English matter? if the function is work
> > > > 
> > > > 
> > > > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> > > 
> > > 
> > > I dont have a better way to name this
> > 
> > 
> > name it EnableOptimizations or something
> 
> I renamed it

i meant in the code

TheFocusMan commented:
> > > > > > > This should have a different name other than fast boost, it's not proper English
> > > > > > 
> > > > > > 
> > > > > > The English matter? if the function is work
> > > > > 
> > > > > 
> > > > > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> > > > 
> > > > 
> > > > I dont have a better way to name this
> > > 
> > > 
> > > name it EnableOptimizations or something
> > 
> > 
> > I renamed it
> 
> i meant in the code

i renamed in the code

terminal-cs commented:
why littleoptimization? that doesn't make too much sense in english either
why not just Optimization?

quajak commented on source/Cosmos.IL2CPU/AppAssembler.cs:
```diff
@@ -1371,7 +1371,7 @@ public void EmitEntrypoint(MethodBase aEntrypoint, MethodBase[] aBootEntries = n
             }
             else
             {
-                X86.IL.Call.DoExecute(Assembler, null, aEntrypoint.DeclaringType.BaseType.GetMethod(CompilerEngine.UseGen3Kernel ? "EntryPoint" : "Start"), null, xCurLabel, CosmosAssembler.EntryPointName + ".AfterStart", DebugEnabled);
```
Why did you change this?

TheFocusMan replied:
To override Kernel.Start

quajak replied:
I see, that is a sensible change. Can you add a sentence to https://github.com/CosmosOS/Cosmos/blob/master/Docs/articles/Kernel/Startup.md/ so that people are aware of the possibility?

quajak reviewed (changes requested):
Thanks for the contribution!

Regarding naming, can you please name this `BootDebugOutput` or similar?

TheFocusMan commented:
> Thanks for the contribution!
> 
> Regarding naming, can you please name this `BootDebugOutput` or similar?

I renamed it,
To get the code work all the three pull requests need to be merged

quajak reviewed (approved):
The change looks good. Please add documentation for the change to the entry method determination. We can merge this when the other two are ready.

TheFocusMan commented:
> The change looks good. Please add documentation for the change to the entry method determination. We can merge this when the other two are ready.

Ok now i added a sentence that explain to people can override Kernel.Start()

quajak commented:
Thanks
